apps/rush-lib/src/cli/taskRunner/TaskRunner.ts

```diff
@@ -151,9 +151,7 @@ export class TaskRunner {
       return taskB.criticalPathLength! - taskA.criticalPathLength!;
     });
```

Comment:
What happens if this throws?

Reply:
Which line are you referring to?

---
In reply to: [185678854](https://github.com/Microsoft/web-build-tools/pull/636#discussion_r185678854) [](ancestors = 185678854)

Reply:
I meant the synchronous part of this function.  When a function returns a promise but starts with a non-promise code block, it creates bad semantics where the function can EITHER reject the promise OR throw a normal exception.  Sometimes this makes it difficult for callers to handle errors correctly.

---
In reply to: [185679126](https://github.com/Microsoft/web-build-tools/pull/636#discussion_r185679126) [](ancestors = 185679126,185678854)